controllers/druid/handler.go

```diff
@@ -391,6 +395,158 @@ func checkCrashStatus(sdk client.Client, m *v1alpha1.Druid) {
+
+	ok := checkIfStsHasEverythingElseSame(sdk, drd, ls, configMapSHA, serviceName, nodeSpecUniqueStr, nodeSpec)
+	if !ok {
+		return
+	}
+
+	// Validate Request, shrinking of pvc not supported
+	// desired size cant be less than current size
+	// in that case re-create sts/pvc which is a user execute manual step
+	if desVolumeClaimTemplateSize.String() < currVolumeClaimTemplateSize.String() {
+		msg := fmt.Sprintf("Request for Shrinking of sts pvc size [sts:%s] in [namespace:%s] is not Supported", sts.Name, sts.Namespace)
+		logger.Info(msg)
+		sendEvent(sdk, drd, v1.EventTypeWarning, "STS_SIZE_FAIL", msg)
+		return
+	}
+
+	// In case size dont match and dessize > currsize, delete the sts using casacde=false / propagation policy set to orphan
+	// The operator on next reconcile shall create the sts with latest changes
+	if desVolumeClaimTemplateSize.String() != currVolumeClaimTemplateSize.String() {
+		msg := fmt.Sprintf("Detected Change in VolumeClaimTemplate Sizes for Statefuleset [%s] in Namespace [%s], desVolumeClaimTemplateSize: [%s], currVolumeClaimTemplateSize: [%s]\n, deleteing STS [%s] with casacde=false]", sts.Name, sts.Namespace, desVolumeClaimTemplateSize.String(), currVolumeClaimTemplateSize.String(), sts.Name)
+		logger.Info(msg)
+		sendEvent(sdk, drd, v1.EventTypeNormal, "STS_SIZE_DETECTED", msg)
+
+		if err := sdk.Delete(context.TODO(), sts, client.PropagationPolicy(metav1.DeletePropagationOrphan)); err != nil {
+			e := fmt.Errorf("failed to Delete [StatefuleSet:%s] due to [%s]", nodeSpecUniqueStr, err.Error())
+			logger.Error(e, e.Error(), "name", drd.Name, "namespace", drd.Namespace)
+			sendEvent(sdk, drd, v1.EventTypeWarning, "DELETE_FAIL", e.Error())
+		} else {
+			msg := fmt.Sprintf("[StatefuleSet:%s] successfully deleted with casacde=false", sts.Name)
+			logger.Info(msg, "name", drd.Name, "namespace", drd.Namespace)
+			sendEvent(sdk, drd, v1.EventTypeNormal, "DELETE_SUCCESS", msg)
+		}
+	}
+
+	// In case size dont match, patch the pvc with the desiredsize from druid CR
+	if desVolumeClaimTemplateSize.String() != pvcSize.String() {
+		for i := range pvc {
+			patch := client.MergeFrom(pvc[i].DeepCopy())
+			pvc[i].Spec.Resources.Requests[v1.ResourceStorage] = desVolumeClaimTemplateSize
+			err := sdk.Patch(context.TODO(), pvc[i], patch)
+			if err != nil {
+				e := fmt.Errorf("Failed to Patch [PVC:%s] due to [%s]", pvc[i].Name, err.Error())
+				logger.Error(e, e.Error(), "name", drd.Name, "namespace", drd.Namespace)
+				sendEvent(sdk, drd, v1.EventTypeWarning, "DELETE_FAIL", e.Error())
+				return
+			} else {
+				msg := fmt.Sprintf("[PVC:%s] successfully Patched with [Size:%s]", pvc[i].Name, pvcSize.String())
+				logger.Info(msg, "name", drd.Name, "namespace", drd.Namespace)
+				sendEvent(sdk, drd, v1.EventTypeNormal, "PATCH_SUCCESS", msg)
+			}
+		}
+	}
+	// after deleting the sts, it takes some time for the sts to come
```

Comment:
do you mean following?  we just deleted the sts, it can't come up by itself... no ?

```suggestion
	// after updating the pvc, it takes some time for the volume expansion to happen
```